Student-Unity/src/main/java/csed/swe/studentunity/DAO/StudentTaskRepo.java

```diff
@@ -10,32 +10,33 @@
 
 public interface StudentTaskRepo extends JpaRepository<StudentTask, StudentTaskId> {
 
-    @Query("SELECT new Task (t.title, t.dueDate, t.telegramLink), new StudentTask(st.status) " +
-            "FROM Task AS t, StudentTask AS st, User AS u " +
-            "WHERE t = st.studentTaskId.task AND u.id = :userId AND st.studentTaskId.user.id = :userId"
+    @Query(value="SELECT t.task_id,t.course_code,t.title,t.due_date,t.telegram_link,st.status\n" +
+            "FROM task AS t, student_task AS st, user AS u \n" +
```

Comment:
you should avoid cross join

Reply:
Why ? 
I think that it is the same If a WHERE clause is added to a cross join, it behaves as an inner join as the WHERE imposes a limiting factor.

I do not know why you said that but to my knowledge the query optimizer do that for you.

You may be right I don't know we can ask ENG El-Naggar about that.
I used the join in that way as I see it more readable

Reply:
ok we can discuss it later